content/guides/user-guides/csi/ceph csi/_index.md

```diff
@@ -0,0 +1,364 @@
+---
+title: CEPH CSI User Guide description: "A user guide for integrating SODA with CEPH CSI Driver via unified Plugin and
+hotpot project"
+weight: 30 disableToc: false tags: ["user guide", "csi", "kubernetes csi", "plug-n-play", "unified plugin"]
+---
+
+# SODA Integration with Vendor CSI Drivers via SODA CSI Plug-N-Play
+
+SODA CSI Plug-N-Play is a mechanism by which users can use the heterogeneous storage vendors in a unified way, Users
+need to define the requirements while creating SODA Profile, Users need to use this profile ID while creating the PVC's
+in the Kubernetes, the soda-csi-provisioner will automatically select the vendor csi-driver and help in provisioning the
+Volumes for Pods.
+
+## Prerequisite
+
+- An installation of Kubernetes (V1.17+)
+- SODA installation (you can refer to quick start guide
+  over [here](https://docs.sodafoundation.io/soda-gettingstarted/installation-using-ansible/)), you need to change the
+  ports of etcd in osdb.yaml while installing SODA on an existing K8s environment as the ports cause conflict with k8s
+  etcd.
+
+## Selecting the vendor CSI driver
+
+You can select the CSI driver supported by SODA Plug-N-Play
+from [here](https://docs.sodafoundation.io/guides/user-guides/nbp/csi-pnp/) Once selected follow the instruction given
+in the document to deploy the CSI driver in K8s.
+
+##### For this example we are selecting the [CEPH-RBD CSI Driver](https://github.com/ceph/ceph-csi).
+
+Following are some additional steps we need to perform for this plugin
+
+* Setup CEPH Client Authentication.
+
+```
+ceph auth get-or-create client.kube osd 'allow rwx pool=osdsrbd' mon 'allow r' -o /etc/ceph/ceph.client.kube.keyring
+```
+
+* Created Data Pool needs to be associated with rbd application.
+
+```
+ceph osd pool application enable osdsrbd rbd
+```
+
+* Initialize the pool to be used by RBD.
+
+```
+rbd pool init osdsrbd
+```
+
+* Adjust the tunables on the ceph cluster.
+
+```
+ceph osd crush tunables hammer
+
+grep -q "^rbd default features" /etc/ceph/ceph.conf || sed -i '/\[global\]/arbd default features = 1' /etc/ceph/ceph.conf
+```
+
+#### Deploy CEPH-CSI configmap.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+data:
+  config.json: |-
+    [
+      {
+        "clusterID": "4ac5251b-114f-4044-9bec-2d27fadad502",
+        "monitors": [
+          "192.168.1.59:6789"
+        ]
+      }
+    ]
+metadata:
+  name: ceph-csi-config
+```
+
+#### Deploy Ceph RBD CSI driver along with soda-csi-provisioner.
+
+```
+kubectl create -f deploy/kubernetes/cephcsi/rbd
+
+kubectl get pods
+NAME                                         READY   STATUS    RESTARTS   AGE
+csi-rbdplugin-6pw9z                          3/3     Running   0          7s
+csi-rbdplugin-provisioner-6b8b9d99fd-x4wn6   7/7     Running   0          7s
+
+```
+
+#### Deploy Secret with userid and userkey of the CEPH user.
+
+kubectl create -f deploy/kubernetes/ceph/secret.yaml
```

Comment:
Use a single line code block.
kubectl create -f deploy/kubernetes/ceph/secret.yaml -> `kubectl create -f deploy/kubernetes/ceph/secret.yaml`

Reply:
@anvithks I have done the changes, please check.